Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -43,16 +43,12 @@
     private static final Logger logger = Logger.getLogger(AzureWebAppMvpModel.class.getName());
 
     public static final String CANNOT_GET_WEB_APP_WITH_ID = "Cannot get Web App with ID: ";
-    private final Map<String, List<ResourceEx<WebApp>>> subscriptionIdToWebApps;
 
     private static final List<WebAppUtils.WebContainerMod> JAVA_8_JAR_CONTAINERS =
         Collections.singletonList(WebAppUtils.WebContainerMod.Java_SE_8);
     private static final List<WebAppUtils.WebContainerMod> JAVA_11_JAR_CONTAINERS = Collections.singletonList(
         WebAppUtils.WebContainerMod.Java_SE_11);
-
-    private AzureWebAppMvpModel() {
-        subscriptionIdToWebApps = new ConcurrentHashMap<>();
-    }
+    public static final String CACHE_SUBSCRIPTION_WEBAPPS = "subscription-webapps";
```

Comment:
Do we also need to add cache for `getById` request?

Reply:
which `getById` method?

Reply:
method like `getWebAppById` to get specific resource

Reply:
it's required if the request is slow, otherwise it's not necessary.